src/Visualization/Python/InterpolateVolumeData.py

```diff
@@ -0,0 +1,324 @@
+            np.array(source_vol.get_tensor_component(obs, name), copy=False)
+            for name in tensor_names
+        ]
+
+        volume_data = []
+        # iterate over elements
+        for grid_name, extent, basis, quadrature in zip(
+                grid_names, extents, bases, quadratures):
+            source_mesh = Mesh[dim](extent, basis_from_string(basis),
+                                    quadrature_from_string(quadrature))
+
+            interpolant = RegularGrid[dim](source_mesh, target_mesh)
+
+            tensor_comps = []
+            offset, length = spectre_h5.offset_and_length_for_grid(
+                grid_name, grid_names, extents)
+            # iterate over tensors
+            for j, tensor in enumerate(tensors):
+                component_data = DataVector(tensor[offset:offset + length],
+                                            copy=False)
+                interpolated_tensor = interpolant.interpolate(component_data)
+                tensor_path = os.path.join(grid_name, tensor_names[j])
+                tensor_comps.append(
+                    TensorComponent(
+                        tensor_path, DataVector(interpolated_tensor,
+                                                copy=False)))
+
+            volume_data.append(
+                ElementVolumeData(target_mesh.extents(), tensor_comps,
+                                  target_mesh.basis(),
+                                  target_mesh.quadrature()))
+        target_vol = target_file.get_vol(target_volume_data)
+        target_vol.write_volume_data(obs, obs_value, volume_data)
+
+    source_file.close()
+    if not target_file is source_file:
+        target_file.close()
+
+
+def parse_args(sys_args):
+    """
+    defines and parses the command line arguments
+
+    Parameters
+    ----------
+    sys_args arguments passed via command line
+
+    Returns
+    -------
+    A dictionary of the parsed command line arguments
+    """
+    parser = argparse.ArgumentParser(
+        description=
+        "Interpolate the tensor components of h5 files to a desired grid")
+
+    parser.add_argument(
+        "--source-file-prefix",
+        required=True,
+        type=str,
+        help="The prefix for the .h5 source files. All files starting with the "
+        "prefix followed by a number will be interpolated.")
+
+    parser.add_argument("--source-subfile-name",
+                        type=str,
+                        required=True,
+                        help="The name of the volume data subfile within the "
+                        "source files in which the data is contained")
+
+    parser.add_argument(
+        "--target-file-prefix",
+        default=None,
+        type=str,
+        help="the prefix for the target files where the interpolated data is "
+        "written. When no target file is specified, the interpolated data is "
+        "written to the corresponding source file in a new volume data subfile."
+    )
+
```

Comment:
(optional) You seem to like blank lines :D I'd delete a lot of them, but up to you